drizzlepac/haputils/mvm_quality_analysis.py

```diff
@@ -0,0 +1,270 @@
+               log_level=logutil.logging.NOTSET):
+    """Report the WCSNAME for each input exposure of an MVM product
+
+    Parameters
+    ----------
+    total_product_list: list of HAP TotalProduct objects, one object per instrument detector
+    (drizzlepac.haputils.Product.TotalProduct)
+
+    json_timestamp: str, optional
+        Universal .json file generation date and time (local timezone) that will be used in the instantiation
+        of the HapDiagnostic object. Format: MM/DD/YYYYTHH:MM:SS (Example: 05/04/2020T13:46:35). If not
+        specified, default value is logical 'None'
+
+    json_time_since_epoch : float
+        Universal .json file generation time that will be used in the instantiation of the HapDiagnostic
+        object. Format: Time (in seconds) elapsed since January 1, 1970, 00:00:00 (UTC). If not specified,
+        default value is logical 'None'
+
+    log_level : int, optional
+        The desired level of verboseness in the log statements displayed on the screen and
+        written to the .log file.  Default value is 'NOTSET'.
+    """
+    log.setLevel(log_level)
+    log.info('\n\n*****     Begin Quality Analysis Test: report_wcsname.     *****\n')
+    aposteriori_list = ['FIT', 'SVM', 'MVM']
+
+    # Generate a separate JSON file for each ExposureProduct object
+    for total_product in total_product_list:
+        detector = total_product.detector
+
+        # Construct the output JSON filename
+        json_filename = '_'.join([total_product.product_basename, 'mvm_wcsname.json'])
+
+        # Set up the diagnostic object
+        diagnostic_obj = du.HapDiagnostic()
+        diagnostic_obj.instantiate_from_hap_obj(total_product,
+                                                data_source='{}.report_wcsname'.format(__taskname__),
+                                                description='WCS information',
+                                                timestamp=json_timestamp,
+                                                time_since_epoch=json_time_since_epoch)
+
+        # Get the WCS for the entire MVM layer
+        metawcs = HSTWCS(total_product.drizzle_filename, ext=1)
+
+        # Loop over all the individual exposures in the list which comprise the layer
+        counter = 0
+        for edp_object in total_product.edp_list:
+
+            # Get the WCSNAME for this specific exposure
+            wcsname = fits.getval(edp_object.full_filename, 'WCSNAME', ext=1)
+
+            # Obtain the value of S_REGION from all SCI extensions in the file
+            sregion = pu.interpret_sregion(edp_object.full_filename, extname='SCI')
+
+            # Convert the S_REGION into a single list containing lists of RA/Dec coordinates
+            sregion_list = []
+            for item in range(len(sregion)):
+                sregion_list.extend(sregion[item].tolist())
+
+            # Capture the RA and Dec coordinates into their own lists
```

Comment:
It may be more efficient and use less code to convert the "sregion_list" to a numpy array, then extract the "ra" and "dec" arrays using indexing syntax such as "ra = sregion_arr[:,0]".  Appending to a list can actually be inefficient.

Reply:
Made the suggested changes here.